### PR TITLE
Fix empty object params bug

### PR DESCRIPTION
**Pull request by jelilat**
## Usage related changes

Closes https://github.com/0xSpaceShard/starknet-devnet-rs/issues/345

## Checklist:

- [x] Applied formatting - `./scripts/format.sh`
- [x] No linter errors - `./scripts/clippy_check.sh`
- [x] No unused dependencies - `./scripts/check_unused_deps.sh`
- [x] Performed code self-review
- [x] Rebased to the last commit of the target branch (or merged it into my branch)
- [ ] Documented the changes
- [x] Linked the issues which this PR resolves
- [ ] Checked the TODO section in README.md if this PR resolves it
- [x] Updated the tests
- [x] All tests are passing - `cargo test`

### CODE REVIEW

File: crates/starknet-devnet-server/src/api/serde_helpers.rs [hunk 1]
```diff
@@ -1,18 +1,36 @@
...
+                if arr.is_empty() {
+                    Ok(())
+                } else {
+                    Err(DeError::custom(format!(
+                        "expected params array with length 0 but got {}",
+                        arr.len()
+                    )))
+                }
+            }
+            _ => Err(DeError::custom("expected empty object or array for params")),
```
Comment (FabijanC) [inline crates/starknet-devnet-server/src/api/serde_helpers.rs:33]: Can be simplified:
```suggestion
            Value::Object(obj) if obj.is_empty() => Ok(()),
            Value::Array(arr) if arr.is_empty() => Ok(()),
            other => Err(DeError::custom(format!(
                "expected empty object or array for params; got: {other:?}"
            ))),
```

File: crates/starknet-devnet-server/src/api/serde_helpers.rs [hunk 2]
```diff
@@ -1,18 +1,36 @@
 /// A module that deserializes `[]` optionally
 pub mod empty_params {
-    use serde::{Deserialize, Deserializer};
+    use serde::{de::Error as DeError, Deserialize, Deserializer};
+    use serde_json::Value;
```
Comment (FabijanC) [inline crates/starknet-devnet-server/src/api/serde_helpers.rs:4]: It's a shame you deleted the content of the PR template, it contains a reminder to format the code using `./scripts/format.sh`. I believe it has a different opinion on the order of imports here.

**Review by FabijanC — CHANGES_REQUESTED** [review]

**Review by marioiordanov — CHANGES_REQUESTED** [review]
circle ci will fail on ./scripts/format.sh

File: crates/starknet-devnet-server/src/api/serde_helpers.rs (hunk 1 above)
Comment (jelilat) [reply crates/starknet-devnet-server/src/api/serde_helpers.rs]: Updated. thanks

File: crates/starknet-devnet-server/src/api/serde_helpers.rs (hunk 2 above)
Comment (jelilat) [reply crates/starknet-devnet-server/src/api/serde_helpers.rs]: I have now followed the template and added tests

**Review by FabijanC — APPROVED** [review]
Thanks for the PR. Looks much better with the test now.

**Comment by jelilat** [comment]
@marioiordanov 

> circle ci will fail on ./scripts/format.sh

Fixed now

**Review by marioiordanov — APPROVED** [review]

**Pull request merged**